src/NuGet.Core/NuGet.Common/ConcurrencyUtilities.cs

```diff
@@ -27,6 +27,28 @@ public static class ConcurrencyUtilities
         private static bool UseDeleteOnClose = RuntimeEnvironmentHelper.IsWindows ||
                                                Environment.GetEnvironmentVariable("NUGET_ConcurrencyUtils_DeleteOnClose") == "1"; // opt-in.
 
+        //////////////////////////////////////////////////////////
+        // Start - Chocolatey Specific Modification
+        //////////////////////////////////////////////////////////
+
+        public async static Task<T> ExecuteWithFileLockedAsync<T>(string filePath,
+            Func<CancellationToken, Task<T>> action,
+            CancellationToken token,
+            string lockFileDirectoryPath)
+        {
+            BasePath = lockFileDirectoryPath;
+
+            var result = await ExecuteWithFileLockedAsync(filePath, action, token);
+
+            BasePath = null;
```

Comment:
Is there any reason we don't do something like this?

```suggestion
            var originalBasePath = BasePath;
            BasePath = lockFileDirectoryPath;

            var result = await ExecuteWithFileLockedAsync(filePath, action, token);

            BasePath = originalBasePath;
```

Reply:
I don't see any reason for explicitly setting it to `null` instead of the original base path, and I would say it makes sense to revert it to the original base path after the execution instead of null.

I'll make the necessary changes and get it pushed up.

Reply:
Just to close the loop here, since originally there wasn't a setter for this property, and since the value of BasePath was established in the getter, when the value was null, I thought it was ok to simply set back to null and original functionality would have been returned.  No objections to the change that is suggested here though.